gnome-initial-setup/pages/live-chooser/eos-setup-live-user

```diff
@@ -49,48 +49,14 @@ GS_SCHEMA = 'org.gnome.software'
 ALLOW_UPDATES = 'allow-updates'
 
 
-class SettingsUpdates(object):
-    def __init__(self, schema):
-        self._settings = Gio.Settings(schema=schema)
-        self._updates = {}
-
-    def get_settings(self):
-        return self._settings
-
-    def set(self, key, value):
-        self._updates[key] = value
-
-    def to_keyfile_str(self):
-        schema_id = self._settings.get_property('schema-id')
-        keyfile_lines = '[{}]\n'.format(schema_id.replace('.', '/'))
-
-        for k, v in self._updates.items():
-            if k is None:
-                continue
-            keyfile_lines += '{}={}\n'.format(k, v.print_(False))
-        return keyfile_lines
-
-    def apply(self):
-        for key, value in self._updates.items():
-            self._settings.set_value(key, value)
-
-
 class SetupLiveUser(object):
     def __init__(self):
-        self.updates = {SHELL_SCHEMA: SettingsUpdates(SHELL_SCHEMA),
-                        GS_SCHEMA: SettingsUpdates(GS_SCHEMA)}
+        self.keyfile = GLib.KeyFile()
 
     def update(self, schema, key, variant):
-        log.info('Updating %s: %s to %s',
-                 schema, key, variant.print_(False))
-        self.updates[schema].set(key, variant)
-
-    def format_keyfile(self):
-        keyfile_lines = []
-        for updates in self.updates.values():
-            keyfile_lines.append(updates.to_keyfile_str())
-
-        return '\n'.join(keyfile_lines)
+        value = variant.print_(False)
+        log.info('Updating %s: %s to %s', schema, key, value)
+        self.keyfile.set_string(schema.replace('.', '/'), key, value)
```

Comment:
Calling `g_key_file_set_string` here with the stringified variant value seems a little dangerous. Is the value from `g_variant_print` always guaranteed to be the representation expected by `GKeyFile`?

Reply:
It's not about what GKeyFile expects, it's about what `dconf(1)` expects. `man 1 dconf` describes the format in passing:

```
       dump
           Dump an entire subpath to stdout. The output is in a keyfile-like
           format, with values in GVariant syntax.
```

[Here's the code that reads them.](https://git.gnome.org/browse/dconf/tree/bin/dconf-update.vala#n119) The [corresponding `dconf dump` code](https://git.gnome.org/browse/dconf/tree/bin/dconf-dump.vala#n1) uses exactly this method to write them.

Reply:
Makes sense - thanks for clarifying